src/serializer/ResidualHeapSerializer.js

```diff
@@ -1153,6 +1211,77 @@ export class ResidualHeapSerializer {
+    let rewrittenAdditionalFunctions: Map<FunctionValue, Array<BabelNodeStatement>> = new Map();
+    let shouldEmitLog = !this.residualHeapValueIdentifiers.collectValToRefCountOnly;
+    let processAdditionalFunctionValuesFn = () => {
+      let additionalFVEffects = this.additionalFunctionValuesAndEffects;
+      if (additionalFVEffects) {
+        for (let [additionalFunctionValue, effects] of additionalFVEffects.entries()) {
+          let [
+            result,
+            generator,
+            modifiedBindings,
+            modifiedProperties: Map<PropertyBinding, void | Descriptor>,
+            createdObjects,
+          ] = effects;
+          let nestedFunctions = new Set([...createdObjects].filter(object => object instanceof FunctionValue));
+          // result -- ignore TODO: return the result from the function somehow
+          // Generator -- visit all entries
+          // Bindings -- only need to serialize bindings if they're captured by some nested function ??
+          //          -- need to apply them and maybe need to revisit functions in ancestors to make sure
+          //          -- we don't overwrite anything they capture
+          //          -- TODO: deal with these properly
+          // PropertyBindings -- visit any property bindings that aren't to createdobjects
+          // CreatedObjects -- should take care of itself
+          this.realm.applyEffects([
+            result,
+            new Generator(this.realm),
+            modifiedBindings,
+            modifiedProperties,
+            createdObjects,
+          ]);
+          // Allows us to emit function declarations etc. inside of this additional
+          // function instead of adding them at global scope
+          // TODO: make sure this generator isn't getting mutated oddly
+          this.additionalFunctionValueNestedFunctions = ((nestedFunctions: any): Set<FunctionValue>);
+          let serializePropertiesAndBindings = () => {
+            let serializedPropertyBindings = this.serializedPropertyBindings;
+            invariant(serializedPropertyBindings);
+            for (let propertyBinding of modifiedProperties.keys()) {
+              let binding: PropertyBinding = ((propertyBinding: any): PropertyBinding);
+              let object = binding.object;
+              if (object instanceof ObjectValue && createdObjects.has(object)) continue;
+              if (object.refuseSerialization) continue;
+              // we've already serialized it from a generator entry.
+              if (serializedPropertyBindings.has(binding)) continue;
+              invariant(object instanceof ObjectValue);
+              this._emitProperty(object, binding.key, binding.descriptor, true);
+            }
+          };
+          let body = this._serializeAdditionalFunction(generator, serializePropertiesAndBindings);
+          invariant(body.length > 0, "An additional function has no body");
```

Comment:
Why is the invariant true that the body has one entry? What happens when I run prepack on an additional function that actually does nothing?